app/services/atmosphere/optimizer.rb

```diff
@@ -14,6 +14,12 @@ def run(hint)
 
     #private
     def satisfy_appliance(appliance)
+
+      action = Action.create { |action|
```

Comment:
Shadowing outer local variable - `action`.<br>Useless assignment to variable - `action`.<br>Avoid using {...} for multi-line blocks.